File: src/models/audit.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use serde_json::Value as JsonValue;
use sqlx::FromRow;
use std::fmt;
use std::net::IpAddr;
use uuid::Uuid;

use crate::models::user::UserRole;

/// 审计事件类型
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
#[sqlx(type_name = "audit_event_type", rename_all = "snake_case")]
pub enum AuditEventType {
    // 用户事件
    UserLogin,
    UserLogout,
    UserRegister,
    UserPasswordChange,
    UserProfileUpdate,
    // 房间事件
    RoomCreate,
    RoomDelete,
    RoomMemberAdd,
    RoomMemberRemove,
    RoomMemberRoleChange,
    // 消息事件
    MessageSend,
    MessageEdit,
    MessageDelete,
    MessageReport,
    // 管理员事件
    AdminUserDisable,
    AdminUserRoleChange,
    AdminUserDelete,
    AdminRoomDelete,
    AdminMessageDelete,
    AdminConfigUpdate,
    // 系统事件
    SystemLoginFailure,
    SystemUnauthorizedAccess,
    SystemRateLimitTriggered,
    // 审计系统事件
    AuditQuery,
    AuditExport,
    AuditStatsQuery,
    AlertQuery,
    AlertRuleUpdate,
    AuditCleanup,
}

impl fmt::Display for AuditEventType {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = match self {
            AuditEventType::UserLogin => "user_login",
            AuditEventType::UserLogout => "user_logout",
            AuditEventType::UserRegister => "user_register",
            AuditEventType::UserPasswordChange => "user_password_change",
            AuditEventType::UserProfileUpdate => "user_profile_update",
            AuditEventType::RoomCreate => "room_create",
            AuditEventType::RoomDelete => "room_delete",
            AuditEventType::RoomMemberAdd => "room_member_add",
            AuditEventType::RoomMemberRemove => "room_member_remove",
            AuditEventType::RoomMemberRoleChange => "room_member_role_change",
            AuditEventType::MessageSend => "message_send",
            AuditEventType::MessageEdit => "message_edit",
            AuditEventType::MessageDelete => "message_delete",
            AuditEventType::MessageReport => "message_report",
            AuditEventType::AdminUserDisable => "admin_user_disable",
            AuditEventType::AdminUserRoleChange => "admin_user_role_change",
            AuditEventType::AdminUserDelete => "admin_user_delete",
            AuditEventType::AdminRoomDelete => "admin_room_delete",
            AuditEventType::AdminMessageDelete => "admin_message_delete",
            AuditEventType::AdminConfigUpdate => "admin_config_update",
            AuditEventType::SystemLoginFailure => "system_login_failure",
            AuditEventType::SystemUnauthorizedAccess => "system_unauthorized_access",
            AuditEventType::SystemRateLimitTriggered => "system_rate_limit_triggered",
            AuditEventType::AuditQuery => "audit_query",
            AuditEventType::AuditExport => "audit_export",
            AuditEventType::AuditStatsQuery => "audit_stats_query",
            AuditEventType::AlertQuery => "alert_query",
            AuditEventType::AlertRuleUpdate => "alert_rule_update",
            AuditEventType::AuditCleanup => "audit_cleanup",
        };
        write!(f, "{}", s)
    }
}

impl AuditEventType {
    /// 获取事件类型的分类
    pub fn category(&self) -> &'static str {
        match self {
            AuditEventType::UserLogin
            | AuditEventType::UserLogout
            | AuditEventType::UserRegister
            | AuditEventType::UserPasswordChange
            | AuditEventType::UserProfileUpdate => "user",
            AuditEventType::RoomCreate
            | AuditEventType::RoomDelete
            | AuditEventType::RoomMemberAdd
            | AuditEventType::RoomMemberRemove
            | AuditEventType::RoomMemberRoleChange => "room",
            AuditEventType::MessageSend
            | AuditEventType::MessageEdit
            | AuditEventType::MessageDelete
            | AuditEventType::MessageReport => "message",
            AuditEventType::AdminUserDisable
            | AuditEventType::AdminUserRoleChange
            | AuditEventType::AdminUserDelete
            | AuditEventType::AdminRoomDelete
            | AuditEventType::AdminMessageDelete
            | AuditEventType::AdminConfigUpdate => "admin",
            AuditEventType::SystemLoginFailure
            | AuditEventType::SystemUnauthorizedAccess
            | AuditEventType::SystemRateLimitTriggered => "system",
            AuditEventType::AuditQuery
            | AuditEventType::AuditExport
            | AuditEventType::AuditStatsQuery
            | AuditEventType::AlertQuery
            | AuditEventType::AlertRuleUpdate
            | AuditEventType::AuditCleanup => "audit",
        }
    }

    /// 获取默认的严重级别
    pub fn default_severity(&self) -> AuditSeverity {
        match self {
            AuditEventType::SystemLoginFailure => AuditSeverity::Warning,
            AuditEventType::SystemUnauthorizedAccess => AuditSeverity::Error,
            AuditEventType::SystemRateLimitTriggered => AuditSeverity::Warning,
            AuditEventType::AdminUserDelete
            | AuditEventType::AdminRoomDelete
            | AuditEventType::AdminMessageDelete => AuditSeverity::Warning,
            _ => AuditSeverity::Info,
        }
    }
}

/// 审计严重级别
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
#[sqlx(type_name = "audit_severity", rename_all = "lowercase")]
pub enum AuditSeverity {
    Info,
    Warning,
    Error,
    Critical,
}

impl fmt::Display for AuditSeverity {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = match self {
            AuditSeverity::Info => "info",
            AuditSeverity::Warning => "warning",
            AuditSeverity::Error => "error",
            AuditSeverity::Critical => "critical",
        };
        write!(f, "{}", s)
    }
}

impl AuditSeverity {
    /// 获取严重级别的优先级数值
    pub fn priority(&self) -> i32 {
        match self {
            AuditSeverity::Info => 0,
            AuditSeverity::Warning => 1,
            AuditSeverity::Error => 2,
            AuditSeverity::Critical => 3,
        }
    }
}

/// 告警状态
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
#[sqlx(type_name = "alert_status", rename_all = "lowercase")]
pub enum AlertStatus {
    New,
    Acknowledged,
    Resolved,
    Ignored,
}

/// 审计日志数据库模型
#[derive(Debug, Clone, FromRow, Serialize)]
pub struct AuditLog {
    pub id: Uuid,
    pub event_type: AuditEventType,
    pub severity: AuditSeverity,
    pub actor_id: Option<Uuid>,
    pub actor_name: Option<String>,
    pub actor_role: Option<UserRole>,
    pub target_type: Option<String>,
    pub target_id: Option<Uuid>,
    pub action: String,
    pub description: String,
    pub metadata: Option<JsonValue>,
    pub status: String,
    pub error_message: Option<String>,
    pub created_at: DateTime<Utc>,
}

/// 审计告警数据库模型
#[derive(Debug, Clone, FromRow, Serialize)]
pub struct AuditAlert {
    pub id: Uuid,
    pub rule_id: Option<Uuid>,
    pub alert_type: String,
    pub severity: AuditSeverity,
    pub title: String,
    pub description: String,
    pub related_logs: Option<Vec<Uuid>>,
    pub source_ip: Option<String>,
    pub affected_user_id: Option<Uuid>,
    pub status: AlertStatus,
    pub acknowledged_by: Option<Uuid>,
    pub acknowledged_at: Option<DateTime<Utc>>,
    pub resolved_by: Option<Uuid>,
    pub resolved_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

/// 告警规则数据库模型
#[derive(Debug, Clone, FromRow, Serialize)]
pub struct AlertRule {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub event_type: Option<AuditEventType>,
    pub condition: JsonValue,
    pub severity: AuditSeverity,
    pub enabled: bool,
    pub cooldown_minutes: i32,
    pub notify_admins: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

/// 创建审计日志请求
#[derive(Debug, Clone, Deserialize)]
pub struct CreateAuditLogRequest {
    pub event_type: AuditEventType,
    pub severity: Option<AuditSeverity>,
    pub actor_id: Option<Uuid>,
    pub actor_role: Option<UserRole>,
    pub target_type: Option<String>,
    pub target_id: Option<Uuid>,
    pub action: String,
    pub description: String,
    pub metadata: Option<AuditMetadata>,
    pub status: Option<String>,
    pub error_message: Option<String>,
}

impl CreateAuditLogRequest {
    /// 创建新的审计日志请求
    pub fn new(
        event_type: AuditEventType,
        action: impl Into<String>,
        description: impl Into<String>,
    ) -> Self {
        Self {
            event_type,
            severity: None,
            actor_id: None,
            actor_role: None,
            target_type: None,
            target_id: None,
            action: action.into(),
            description: description.into(),
            metadata: None,
            status: Some("success".to_string()),
            error_message: None,
        }
    }

    /// 设置操作者信息
    pub fn with_actor(mut self, user_id: Uuid, role: UserRole) -> Self {
        self.actor_id = Some(user_id);
        self.actor_role = Some(role);
        self
    }

    /// 设置目标信息
    pub fn with_target(mut self, target_type: impl Into<String>, target_id: Uuid) -> Self {
        self.target_type = Some(target_type.into());
        self.target_id = Some(target_id);
        self
    }

    /// 设置元数据
    pub fn with_metadata(mut self, metadata: AuditMetadata) -> Self {
        self.metadata = Some(metadata);
        self
    }

    /// 设置严重级别
    pub fn with_severity(mut self, severity: AuditSeverity) -> Self {
        self.severity = Some(severity);
        self
    }

    /// 设置为失败状态
    pub fn with_error(mut self, error_message: impl Into<String>) -> Self {
        self.status = Some("failure".to_string());
        self.error_message = Some(error_message.into());
        self.severity = Some(AuditSeverity::Error);
        self
    }

    /// 获取严重级别（使用默认值或指定值）
    pub fn severity(&self) -> AuditSeverity {
        self.severity
            .clone()
            .unwrap_or_else(|| self.event_type.default_severity())
    }
}

/// 审计元数据
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AuditMetadata {
    pub ip_address: Option<IpAddr>,
    pub user_agent: Option<String>,
    pub request_path: Option<String>,
    pub request_method: Option<String>,
    pub request_params: Option<JsonValue>,
    pub extra: Option<JsonValue>,
}

impl AuditMetadata {
    /// 创建新的元数据
    pub fn new() -> Self {
        Self {
            ip_address: None,
            user_agent: None,
            request_path: None,
            request_method: None,
            request_params: None,
            extra: None,
        }
    }

    /// 设置IP地址
    pub fn with_ip(mut self, ip: IpAddr) -> Self {
        self.ip_address = Some(ip);
        self
    }

    /// 设置User-Agent
    pub fn with_user_agent(mut self, ua: impl Into<String>) -> Self {
        self.user_agent = Some(ua.into());
        self
    }

    /// 设置请求信息
    pub fn with_request(
        mut self,
        method: impl Into<String>,
        path: impl Into<String>,
        params: Option<JsonValue>,
    ) -> Self {
        self.request_method = Some(method.into());
        self.request_path = Some(path.into());
        self.request_params = params;
        self
    }

    /// 设置额外数据
    pub fn with_extra(mut self, extra: JsonValue) -> Self {
        self.extra = Some(extra);
        self
    }
}

impl Default for AuditMetadata {
    fn default() -> Self {
        Self::new()
    }
}

/// 审计日志查询参数
#[derive(Debug, Clone, Deserialize)]
pub struct AuditLogQuery {
    pub event_type: Option<AuditEventType>,
    pub severity: Option<AuditSeverity>,
    pub actor_id: Option<Uuid>,
    pub target_id: Option<Uuid>,
    pub target_type: Option<String>,
    pub status: Option<String>,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

impl Default for AuditLogQuery {
    fn default() -> Self {
        Self {
            event_type: None,
            severity: None,
            actor_id: None,
            target_id: None,
            target_type: None,
            status: None,
            start_time: None,
            end_time: None,
            limit: Some(50),
            offset: Some(0),
        }
    }
}

/// 审计日志列表响应
#[derive(Debug, Clone, Serialize)]
pub struct AuditLogListResponse {
    pub logs: Vec<AuditLog>,
    pub total: i64,
    pub limit: i64,
    pub offset: i64,
}

/// 告警查询参数
#[derive(Debug, Clone, Deserialize)]
pub struct AlertQuery {
    pub status: Option<AlertStatus>,
    pub severity: Option<AuditSeverity>,
    pub alert_type: Option<String>,
    pub affected_user_id: Option<Uuid>,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

impl Default for AlertQuery {
    fn default() -> Self {
        Self {
            status: None,
            severity: None,
            alert_type: None,
            affected_user_id: None,
            start_time: None,
            end_time: None,
            limit: Some(50),
            offset: Some(0),
        }
    }
}

/// 告警列表响应
#[derive(Debug, Clone, Serialize)]
pub struct AlertListResponse {
    pub alerts: Vec<AuditAlert>,
    pub total: i64,
    pub limit: i64,
    pub offset: i64,
}

/// 更新告警状态请求
#[derive(Debug, Clone, Deserialize)]
pub struct UpdateAlertStatusRequest {
    pub status: AlertStatus,
}

/// 更新告警规则请求
#[derive(Debug, Clone, Deserialize)]
pub struct UpdateAlertRuleRequest {
    pub name: Option<String>,
    pub description: Option<String>,
    pub condition: Option<JsonValue>,
    pub severity: Option<AuditSeverity>,
    pub enabled: Option<bool>,
    pub cooldown_minutes: Option<i32>,
    pub notify_admins: Option<bool>,
}

/// 创建告警请求
#[derive(Debug, Clone)]
pub struct CreateAlertRequest {
    pub rule_id: Option<Uuid>,
    pub alert_type: String,
    pub severity: AuditSeverity,
    pub title: String,
    pub description: String,
    pub related_logs: Vec<Uuid>,
    pub source_ip: Option<String>,
    pub affected_user_id: Option<Uuid>,
}

/// 审计统计信息
#[derive(Debug, Clone, Serialize)]
pub struct AuditStats {
    pub total_logs: i64,
    pub today_logs: i64,
    pub week_logs: i64,
    pub month_logs: i64,
    pub logs_by_severity: Vec<SeverityCount>,
    pub logs_by_event_type: Vec<EventTypeCount>,
    pub logs_by_day: Vec<DailyCount>,
    pub alerts_count: i64,
    pub new_alerts_count: i64,
}

/// 严重级别统计
#[derive(Debug, Clone, Serialize)]
pub struct SeverityCount {
    pub severity: AuditSeverity,
    pub count: i64,
}

/// 事件类型统计
#[derive(Debug, Clone, Serialize)]
pub struct EventTypeCount {
    pub event_type: AuditEventType,
    pub count: i64,
}

/// 每日统计
#[derive(Debug, Clone, Serialize)]
pub struct DailyCount {
    pub date: String,
    pub count: i64,
}

/// 审计统计查询参数
#[derive(Debug, Clone, Deserialize)]
pub struct AuditStatsQuery {
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
}

/// 审计日志导出查询参数
#[derive(Debug, Clone, Deserialize)]
pub struct AuditLogExportQuery {
    pub event_type: Option<AuditEventType>,
    pub severity: Option<AuditSeverity>,
    pub actor_id: Option<Uuid>,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub format: ExportFormat,
}

/// 导出格式
#[derive(Debug, Clone, Deserialize, Default)]
#[serde(rename_all = "lowercase")]
pub enum ExportFormat {
    #[default]
    Json,
    Csv,
}

/// 告警规则响应
#[derive(Debug, Clone, Serialize)]
pub struct AlertRuleResponse {
    pub rules: Vec<AlertRule>,
}

/// 单个告警规则详情响应
#[derive(Debug, Clone, Serialize)]
pub struct AlertRuleDetailResponse {
    pub rule: AlertRule,
}

File: client/admin/src/api/audit.ts
import { http } from './request'
import type { ApiResponse } from '@/types'

// ==================== 审计日志类型 ====================

/**
 * 审计日志
 */
export interface AuditLog {
  /** 日志ID */
  id: string
  /** 事件类型 */
  event_type: string
  /** 严重级别 */
  severity: 'info' | 'warning' | 'error' | 'critical'
  /** 操作者ID */
  actor_id: string | null
  /** 操作者名称 */
  actor_name: string | null
  /** 操作者角色 */
  actor_role: string | null
  /** 目标类型 */
  target_type: string | null
  /** 目标ID */
  target_id: string | null
  /** 操作动作 */
  action: string
  /** 描述 */
  description: string
  /** 元数据 */
  metadata: Record<string, unknown> | null
  /** 状态 */
  status: string
  /** 错误信息 */
  error_message: string | null
  /** 创建时间 */
  created_at: string
}

/**
 * 审计日志查询参数
 */
export interface AuditLogQueryParams {
  /** 事件类型 */
  event_type?: string
  /** 严重级别 */
  severity?: string
  /** 操作者ID */
  actor_id?: string
  /** 目标类型 */
  target_type?: string
  /** 状态 */
  status?: string
  /** 开始时间 */
  start_time?: string
  /** 结束时间 */
  end_time?: string
  /** 每页数量 */
  limit?: number
  /** 偏移量 */
  offset?: number
  [key: string]: unknown
}

/**
 * 审计日志列表响应
 */
export interface AuditLogListData {
  /** 日志列表 */
  logs: AuditLog[]
  /** 总条数 */
  total: number
  /** 每页数量 */
  limit: number
  /** 偏移量 */
  offset: number
}

// ==================== 安全告警类型 ====================

/**
 * 用户信息（简化）
 */
export interface AlertUserInfo {
  /** 用户ID */
  id: string
  /** 用户名 */
  username: string
  /** 头像URL */
  avatar_url: string | null
}

/**
 * 安全告警
 */
export interface AuditAlert {
  /** 告警ID */
  id: string
  /** 规则ID */
  rule_id: string | null
  /** 告警类型 */
  alert_type: string
  /** 严重级别 */
  severity: 'info' | 'warning' | 'error' | 'critical'
  /** 标题 */
  title: string
  /** 描述 */
  description: string
  /** 关联日志ID列表 */
  related_logs: string[] | null
  /** 来源IP */
  source_ip: string | null
  /** 受影响用户 */
  affected_user: AlertUserInfo | null
  /** 状态 */
  status: 'new' | 'acknowledged' | 'resolved' | 'ignored'
  /** 确认人 */
  acknowledged_by: AlertUserInfo | null
  /** 确认时间 */
  acknowledged_at: string | null
  /** 解决人 */
  resolved_by: AlertUserInfo | null
  /** 解决时间 */
  resolved_at: string | null
  /** 创建时间 */
  created_at: string
  /** 更新时间 */
  updated_at: string
}

/**
 * 告警查询参数
 */
export interface AlertQueryParams {
  /** 状态 */
  status?: string
  /** 严重级别 */
  severity?: string
  /** 告警类型 */
  alert_type?: string
  /** 受影响用户ID */
  affected_user_id?: string
  /** 开始时间 */
  start_time?: string
  /** 结束时间 */
  end_time?: string
  /** 每页数量 */
  limit?: number
  /** 偏移量 */
  offset?: number
  [key: string]: unknown
}

/**
 * 告警列表响应
 */
export interface AlertListData {
  /** 告警列表 */
  alerts: AuditAlert[]
  /** 总条数 */
  total: number
  /** 每页数量 */
  limit: number
  /** 偏移量 */
  offset: number
}

/**
 * 更新告警状态请求
 */
export interface UpdateAlertStatusRequest {
  /** 新状态 */
  status: 'acknowledged' | 'resolved' | 'ignored'
}

// ==================== 告警规则类型 ====================

/**
 * 告警规则
 */
export interface AlertRule {
  /** 规则ID */
  id: string
  /** 规则名称 */
  name: string
  /** 描述 */
  description: string | null
  /** 事件类型 */
  event_type: string | null
  /** 条件配置 */
  condition: Record<string, unknown>
  /** 严重级别 */
  severity: 'info' | 'warning' | 'error' | 'critical'
  /** 是否启用 */
  enabled: boolean
  /** 冷却时间（分钟） */
  cooldown_minutes: number
  /** 是否通知管理员 */
  notify_admins: boolean
  /** 创建时间 */
  created_at: string
  /** 更新时间 */
  updated_at: string
}

/**
 * 更新告警规则请求
 */
export interface UpdateAlertRuleRequest {
  /** 规则名称 */
  name?: string
  /** 描述 */
  description?: string | null
  /** 条件配置 */
  condition?: Record<string, unknown>
  /** 严重级别 */
  severity?: 'info' | 'warning' | 'error' | 'critical'
  /** 是否启用 */
  enabled?: boolean
  /** 冷却时间（分钟） */
  cooldown_minutes?: number
  /** 是否通知管理员 */
  notify_admins?: boolean
}

/**
 * 告警规则列表响应
 */
export interface AlertRuleListData {
  /** 规则列表 */
  rules: AlertRule[]
}

// ==================== 审计统计类型 ====================

/**
 * 严重级别统计
 */
export interface SeverityCount {
  /** 严重级别 */
  severity: string
  /** 数量 */
  count: number
}

/**
 * 事件类型统计
 */
export interface EventTypeCount {
  /** 事件类型 */
  event_type: string
  /** 数量 */
  count: number
}

/**
 * 每日统计
 */
export interface DailyCount {
  /** 日期 */
  date: string
  /** 数量 */
  count: number
}

/**
 * 审计统计信息
 */
export interface AuditStats {
  /** 总日志数 */
  total_logs: number
  /** 今日日志数 */
  today_logs: number
  /** 本周日志数 */
  week_logs: number
  /** 本月日志数 */
  month_logs: number
  /** 按严重级别统计 */
  logs_by_severity: SeverityCount[]
  /** 按事件类型统计 */
  logs_by_event_type: EventTypeCount[]
  /** 按天统计 */
  logs_by_day: DailyCount[]
  /** 告警总数 */
  alerts_count: number
  /** 新增告警数 */
  new_alerts_count: number
}

/**
 * 审计统计查询参数
 */
export interface AuditStatsQueryParams {
  /** 开始时间 */
  start_time?: string
  /** 结束时间 */
  end_time?: string
  [key: string]: unknown
}

/**
 * 导出格式
 */
export type ExportFormat = 'json' | 'csv'

/**
 * 审计日志导出参数
 */
export interface AuditExportParams {
  /** 事件类型 */
  event_type?: string
  /** 严重级别 */
  severity?: string
  /** 操作者ID */
  actor_id?: string
  /** 开始时间 */
  start_time?: string
  /** 结束时间 */
  end_time?: string
  /** 导出格式 */
  format: ExportFormat
  [key: string]: unknown
}

// ==================== API 接口 ====================

/**
 * 构建查询字符串
 * @param params 查询参数
 * @returns URL 查询字符串
 */
function buildQueryString(params: Record<string, unknown>): string {
  const queryParams = new URLSearchParams()
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== '') {
      queryParams.append(key, String(value))
    }
  })
  const query = queryParams.toString()
  return query ? `?${query}` : ''
}

/**
 * 审计系统 API
 */
export const auditApi = {
  /**
   * 获取审计日志列表
   * @param params 查询参数
   * @returns 审计日志列表
   */
  getAuditLogs(params: AuditLogQueryParams = {}): Promise<ApiResponse<AuditLogListData>> {
    return http.get<AuditLogListData>(`/admin/audit/logs${buildQueryString(params)}`)
  },

  /**
   * 获取审计日志详情
   * @param id 日志ID
   * @returns 审计日志详情
   */
  getAuditLogDetail(id: string): Promise<ApiResponse<AuditLog>> {
    return http.get<AuditLog>(`/admin/audit/logs/${id}`)
  },

  /**
   * 获取审计统计信息
   * @param params 查询参数
   * @returns 审计统计
   */
  getAuditStats(params: AuditStatsQueryParams = {}): Promise<ApiResponse<AuditStats>> {
    return http.get<AuditStats>(`/admin/audit/stats${buildQueryString(params)}`)
  },

  /**
   * 导出审计日志
   * @param params 导出参数
   * @returns Blob 数据
   */
  async exportAuditLogs(params: AuditExportParams): Promise<Blob> {
    const query = buildQueryString(params as Record<string, unknown>)
    const response = await fetch(
      `${import.meta.env.VITE_API_BASE_URL || 'http://localhost:8080/api/v1'}/admin/audit/export${query}`,
      {
        method: 'GET',
        headers: {
          Authorization: `Bearer ${localStorage.getItem('access_token') || ''}`,
        },
      }
    )
    if (!response.ok) {
      throw new Error('导出失败')
    }
    return response.blob()
  },

  /**
   * 获取安全告警列表
   * @param params 查询参数
   * @returns 告警列表
   */
  getAlerts(params: AlertQueryParams = {}): Promise<ApiResponse<AlertListData>> {
    return http.get<AlertListData>(`/admin/audit/alerts${buildQueryString(params)}`)
  },

  /**
   * 更新告警状态
   * @param id 告警ID
   * @param data 状态更新请求
   * @returns 更新后的告警
   */
  updateAlertStatus(id: string, data: UpdateAlertStatusRequest): Promise<ApiResponse<AuditAlert>> {
    return http.put<AuditAlert>(`/admin/audit/alerts/${id}/status`, data)
  },

  /**
   * 获取告警规则列表
   * @returns 规则列表
   */
  getAlertRules(): Promise<ApiResponse<AlertRuleListData>> {
    return http.get<AlertRuleListData>('/admin/audit/rules')
  },

  /**
   * 更新告警规则
   * @param id 规则ID
   * @param data 规则更新请求
   * @returns 更新后的规则
   */
  updateAlertRule(id: string, data: UpdateAlertRuleRequest): Promise<ApiResponse<AlertRule>> {
    return http.put<AlertRule>(`/admin/audit/rules/${id}`, data)
  },
}
